React setup

margotTenenbaum opened the pull request:
@Nick-Zabalza @GrumpyCat0709 
Hi all,
This branch covers a very basic React/Webpack setup, in addition to building out the two server routes and associated db helper functions needed for this service. The branch name is a bit of a misnomer, as the bulk of this PR is really the new server routes. The PR includes additional Jest unit tests to cover the new db helper functions, and an updated README with instructions on how to start the server and access the two routes. I have not created tests for the React components yet, as those will be built out in a subsequent PR and there would not be much to test at this point.

Please let me know what you think!

Nick-Zabalza commented:
@margotTenenbaum 

Hi Elizabeth,

No worries on the misnamed branch. Most of your code looks good but I wasn't able to fully test it due to the hard stop I reached. This hard stop also reveals an issue with your unit tests.

Basically, I followed the instructions in your ReadMe and got to the point of running the tests. They all passed. Then I decided to try using the service links in the browser and at that point I got nothing back. Which made me realize I had no data in the my MongoDB instance for your database.

This experience exposes the following issues:
1. No instructions for seeding the database in your ReadMe.
2. I tried looking in your package.json file for a script command I could run to seed the DB. Didn't find one.
3. Your tests managed to pass even with this major oversight. If I recall correctly, you do have some hardcoded example data and so I suggest writing tests that try to retrieve that data and then compares if the data matches what is hardcoded in your seed file.

Another issue I noticed before I encountered all these issues was that none of your tests seemed to test your server endpoints. Which means you have no unit tests for your express server. Considering point 3, you can combine these two issues when making new tests, if you want.

For all of the above issues, I need them remedied before I can approve this PR. One minor issue that isn't required: I noticed a number of console.log statements that aren't necessary in the final product. Yes, I understand they help with development and, ideally, they will be removed at the end. But i looked on your Trello and didn't see a ticket for "Remove extraneous console statements". I made one on mine and suggest you do the same. But you are free to do as you wish on this issue.

margotTenenbaum commented:
Hi @Nick-Zabalza,
Good catch! 
1. package.json and the README have been updated with a seed script and seeding instructions.
2. There is now a unit test to confirm that the database has been seeded.
3. There are now unit tests testing the server endpoints.
4. I have cleaned up a few of the console.log statements, but left many of them as I find them useful. I have added a Trello ticket at your suggestion to revisit them at a later time.

Nick-Zabalza commented:
@margotTenenbaum 

Looks like you are good to go, Elizabeth. Good work!
